library/ssl_tls.c

```diff
@@ -1291,13 +1301,23 @@ void mbedtls_ssl_conf_cert_profile( mbedtls_ssl_config *conf,
     conf->cert_profile = profile;
 }
 
+static void ssl_key_cert_free( mbedtls_ssl_key_cert *key_cert );
+
 /* Append a new keycert entry to a (possibly empty) list */
 static int ssl_append_key_cert( mbedtls_ssl_key_cert **head,
                                 mbedtls_x509_crt *cert,
                                 mbedtls_pk_context *key )
 {
     mbedtls_ssl_key_cert *new_cert;
 
+    if( cert == NULL )
```

Comment:
Unless I'm mistaken, we don't have tests covering that. Testing it via `ssl_server2` seems pretty awkard, but I'm thinking an ad-hoc unit test in `test_suite_ssl` would probably work well.

Reply:
Still todo.  Alternatively, this could be removed from this PR, since `mbedtls_ssl_conf_cert_cb()` allows setting certificates to be deferred until the callback, which is for certificate selection.

Reply:
:rocket: To test, I set and clear and (again) set cert in `mbedtls_endpoint_certificate_init()` in `tests/suites/test_suite_ssl.function`